src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting_NetFx/DocumentFormattingTest.cs

```diff
@@ -18,6 +18,52 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting;
 public class DocumentFormattingTest(FormattingTestContext context, HtmlFormattingFixture fixture, ITestOutputHelper testOutput)
     : FormattingTestBase(context, fixture.Service, testOutput), IClassFixture<FormattingTestContext>
 {
+    private readonly bool _useNewFormattingEngine = context.UseNewFormattingEngine;
+
+    [FormattingTestFact]
+    public async Task EmptyDocument()
+    {
+        await RunFormattingTestAsync(
+            input: "",
+            expected: "");
+    }
+
+    [FormattingTestFact]
+    public async Task AllWhitespaceDocument()
+    {
+        // The Html formatter shrinks this down to one line
+        await RunFormattingTestAsync(
+            input: """
+
+                
+                
+
+            """,
+            expected: """
+
+            """);
+    }
+
+    [FormattingTestFact]
+    public async Task StartsWithWhitespace()
```

Comment:
For completeness do we need an `EndsWithWhitespace`? I Guess it's covered here (and probably elsewhere)

Reply:
I'm happy to add one, but my approach for creating new tests was simply pure reaction. Either there were quirks/bugs I found during development, or conditions I was adding that I wanted to validate. So if there is no `EndsWithWhitespace`, it means there was nothing odd about the tree when it came to formatting it :)

Reply:
I'd say just add it for now. If nothing else serves as a regression test in case the tree changes. The tests are great, run quick, and provide a baseline comfort for changing both the syntax tree in the compiler and formatting behavior in tooling. I'd just err on more coverage even if it's not proven to be needed

Reply:
> run quick

They ran a lot quicker before I made them all run 8 times 😁